test/unit/canvas.js

```diff
@@ -790,6 +790,36 @@
     canvas.remove(triangle);
   });
 
+  QUnit.test('findTarget with perPixelTargetFind in nested group', function(assert) {
+    assert.ok(typeof canvas.findTarget === 'function');
+    var triangle = makeTriangle({ left: 0, top: 0 }),
+        target,
+        group1 = new fabric.Group([triangle], {
+          subTargetCheck: true
+        }),
+        group2 = new fabric.Group([group1], {
+          subTargetCheck: true
+        });
+
+    canvas.add(group2);
```

Comment:
While this test is ok, the generated group is too simple.
can we use something like this:

```javascript
var circle = new fabric.Circle({ radius: 30, top: 30, left: 30, fill: 'yellow'});

var circle2 = new fabric.Ellipse({ rx: 10, ry: 20, top: 40, left: 0, fill: 'blue'});

var circle3 = new fabric.Circle({ scaleX: 2, scaleY: 2, radius: 10, top: 120, left: -20, fill: 'red'});

var rect = new fabric.Rect({ width: 100, height: 80, top: 50, left: 60, fill: 'purple' });

var group = new fabric.Group([circle, circle2], { scaleX: 3, angle: 45, opacity: 0.5 });

var group2 = new fabric.Group([group, rect, circle3], { opacity: 0.5, top: 20, left: 20, scaleX: 0.8, scaleY: 1.2 });

canvas.add(group2);
```
that looks like this:

![image](https://user-images.githubusercontent.com/1194048/46387423-f0c3fb00-c6c6-11e8-9014-b60297dcdf9f.png)

and test edge cases? we should be able to succesfully target all the shapes.